### PR TITLE
fix potential mem leak

### PR DESCRIPTION
**Pull request by hana-alice**
Re: #https://github.com/cocos/3d-tasks/issues/14034 

### Changelog
fix potential memleak, actually MemoryAllocator is a static object so this can't be called even restart, no idea why xcode detect it.
*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by hana-alice** [comment]
@cocos-robot npm run test case

File: native/cocos/base/threading/MessageQueue.cpp [hunk 1]
```diff
@@ -76,6 +76,16 @@ void MessageQueue::MemoryAllocator::freeByUser(MessageQueue *const mainMessageQu
...
 }
 
+ MessageQueue::MemoryAllocator::~MemoryAllocator() noexcept {
+    uint8_t *chunk = nullptr;
+     while (_chunkCount.load(std::memory_order_acquire) != 0) {
+         if(_chunkPool.try_dequeue(chunk)) {
+             free(chunk);
+             _chunkCount.fetch_sub(1, std::memory_order_acq_rel);
+         }
+     }
```
Comment (minggo) [inline native/cocos/base/threading/MessageQueue.cpp]: I think only need load one time, and free all chunks, as it is a destructor, so can modify like
```c++
uint32_t chunkCount = _chunkCount.load(std::memory_order_acquire);
while (chunkCount--) {
    if(_chunkPool.try_dequeue(chunk)) {
             free(chunk);
         }
} 
```

Comment (minggo) [reply native/cocos/base/threading/MessageQueue.cpp]: And i think can even change to
```c++
while (_chunkPool.try_dequeue(chunk)) {
             free(chunk);
} 
```

**Comment by hana-alice** [comment]
@cocos-robot run test case

File: native/cocos/base/threading/MessageQueue.cpp [hunk 2]
```diff
@@ -76,6 +76,14 @@ void MessageQueue::MemoryAllocator::freeByUser(MessageQueue *const mainMessageQu
     mainMessageQueue->kick();
 }
 
+ MessageQueue::MemoryAllocator::~MemoryAllocator() noexcept {
+    uint8_t *chunk = nullptr;
+    while (_chunkPool.try_dequeue(chunk)) {
+        free(chunk);
+        _chunkCount.fetch_sub(1, std::memory_order_acq_rel);
```
Comment (minggo) [inline native/cocos/base/threading/MessageQueue.cpp:87]: It is ok to write code like this. But i think it is not needed in destructor.

**Review by minggo — APPROVED** [review]

**Comment by novain1102** [comment]
Related issue: https://github.com/cocos/cocos-engine/issues/12839

**Review by minggo — APPROVED** [review]

**Pull request merged**